test/retry.js

```diff
@@ -52,43 +43,41 @@ describe('agenda', function() {
     });
   });
 
-  afterEach(function(done) {
-    setTimeout(function() {
-      jobs.stop(function() {
-        clearJobs(function() {
-          mongo.close(function() {
+  afterEach(done => {
+    setTimeout(() => {
+      jobs.stop(() => {
+        clearJobs(() => {
+          mongo.close(() => {
             jobs._mdb.close(done);
           });
         });
       });
     }, 50);
   });
 
-  describe('Retry', function () {
-    it('should retry a job', function(done) {
-      var shouldFail = true;
-      jobs.define('a job', function (job, done) {
-        if(shouldFail) {
-          shouldFail = false;
-          return done(new Error('test failure'));
-        }
-        done();
-      });
+  it('should retry a job', done => {
+    let shouldFail = true;
+    jobs.define('a job', (job, done) => {
+      if (shouldFail) {
+        shouldFail = false;
+        return done(new Error('test failure'));
+      }
+      done();
+    });
 
-      jobs.on('fail:a job', function (err, job) {
-        job
-          .schedule('now')
-          .save();
-      });
+    jobs.on('fail:a job', (err, job) => {
+      if (err) {
+        // Do nothing as this is expected to fail.
```

Comment:
My thoughts exactly.

I'll switch from mocha + expect.js + blanket to [avajs](https://github.com/avajs/ava) which will get us using the latest testing and coverage setup. It has something exactly for this.